src/material/expansion/testing/expansion-harness.ts

```diff
@@ -0,0 +1,127 @@
+  private _title = this.locatorForOptional('.mat-expansion-panel-header-title');
+  private _description = this.locatorForOptional('.mat-expansion-panel-header-description');
+  private _expansionIndicator = this.locatorForOptional('.mat-expansion-indicator');
+  private _content = this.locatorFor(EXPANSION_PANEL_CONTENT_SELECTOR);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for an expansion-panel
+   * with specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `title` finds an expansion-panel with a specific title text.
+   *   - `description` finds an expansion-panel with a specific description text.
+   *   - `expanded` finds an expansion-panel that is currently expanded.
+   *   - `disabled` finds an expansion-panel that is disabled.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ExpansionPanelHarnessFilters = {}):
+      HarnessPredicate<MatExpansionPanelHarness> {
+    return new HarnessPredicate(MatExpansionPanelHarness, options)
+        .addOption(
+            'title', options.title,
+            (harness, title) => HarnessPredicate.stringMatches(harness.getTitle(), title))
+        .addOption(
+            'description', options.description,
+            (harness, description) =>
+                HarnessPredicate.stringMatches(harness.getDescription(), description))
+        .addOption(
+            'content', options.content,
+            (harness, content) => HarnessPredicate.stringMatches(harness.getTextContent(), content))
+        .addOption(
+            'expanded', options.expanded,
+            async (harness, expanded) => (await harness.isExpanded()) === expanded)
+        .addOption(
+            'disabled', options.disabled,
+            async (harness, disabled) => (await harness.isDisabled()) === disabled);
+  }
+
+  /** Whether the panel is expanded. */
+  async isExpanded(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-expanded');
+  }
+
+  /**
+   * Gets the title text of the panel.
+   * @returns Title text or `null` if no title is set up.
+   */
+  async getTitle(): Promise<string|null> {
+    const titleEl = await this._title();
+    return titleEl ? titleEl.text() : null;
+  }
+
+  /**
+   * Gets the description text of the panel.
+   * @returns Description text or `null` if no description is set up.
+   */
+  async getDescription(): Promise<string|null> {
+    const descriptionEl = await this._description();
+    return descriptionEl ? descriptionEl.text() : null;
+  }
+
+  /** Whether the panel is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return await (await this._header()).getAttribute('aria-disabled') === 'true';
+  }
+
+  /**
+   * Toggles the expanded state of the panel by clicking on the panel
+   * header. This method will not work if the panel is disabled.
+   */
+  async toggle(): Promise<void> {
```

Comment:
In addition to `toggle`, I would add methods for `expand` and `collapse` that no-op if the panel is already expanded/collapsed

Reply:
Done.